### PR TITLE
test: Add TAP TestLayout for Screen/Layout scale math

### LINKED ISSUE #3083: Add TAP TestLayout for Screen/Layout scale math
## Summary

Add a TAP unit test for `Layout::Initialise()` and the `Layout::` scale helpers (`Scale`, `PtScale`, `VptScale`, `FontScale`, control heights, hit radius).

This is **DPI / scale math**, not dialog packing (`RowFormWidget` / `WidgetDialog`). Those still need fonts and live windows and stay out of `make check`.

## Why

`Layout::` decides UI size from screen pixels, DPI, `ui_scale`, and small-screen / landscape. A wrong factor shows up everywhere (fonts, buttons, hit targets) and is easy to regress. There is no dedicated test; `TestWrapText` only assigns `Layout::scale_1024` by hand.

## Suggested approach

- New `test/src/TestLayout.cpp` registered in `build/test.mk` (same TAP style as other `Test*` binaries).
- Call `Layout::Initialise()` with a fake `PixelSize` and `custom_dpi` (avoids a real display). Cover a few cases, for example:
  - 240×320 @ 96 DPI, `ui_scale=100` (base)
  - landscape vs portrait (`landscape` flag)
  - square vs non-square (`scale_1024` uses 320 vs 240)
  - small-screen path (`IsSmallScreen`: short edge &lt; 5 inch)
  - `ui_scale` other than 100 (font scale)
  - `HasTouchScreen()` vs not (max control height, `hit_radius`)
- Assert `Scale(240)`, `PtScale` / `FontScale` / `GetMinimumControlHeight()` / `GetHitRadius()` against the formulas in `src/Screen/Layout.cpp`.
- Do **not** instantiate widgets, canvases, or dialogs.

`TestWrapText` already links `Screen/Layout.cpp`; reuse that pattern rather than a new harness.

## Out of scope

Automated tests of `RowFormWidget::UpdateLayout()`, `WndForm`, or pixel-perfect dialog geometry. Interactive checks stay on `Run*` tools.

## References

- `src/Screen/Layout.hpp`, `src/Screen/Layout.cpp`
- `test/src/TestWrapText.cpp` (existing `Layout::` globals)
- `doc/policy.rst` / TAP tests for UI-free logic

### PR DESCRIPTION
**Pull request by lordfolken**
## Summary
- Add a Display-free `Layout::Initialise(size, dpi, ui_scale, has_touch)` so scale math can be tested without X11/Wayland.
- Add TAP `TestLayout` for base, landscape, square, 5-inch, `ui_scale`, touch, and anisotropic DPI, plus a few literal goldens for the 240 vs 320 divisor.
- Closes #3083.

## Test plan
- [x] `make -j$(nproc) TARGET=UNIX USE_CCACHE=y output/UNIX/bin/TestLayout && output/UNIX/bin/TestLayout` (338 TAP checks)
- [ ] `make -j$(nproc) TARGET=UNIX USE_CCACHE=y check`
- [ ] CI `check` on this PR (UNIX VFB)

## Summary by CodeRabbit

- **New Features**
  - Layouts can now be initialized using explicit screen dimensions, display DPI, UI scale, and touch capability, improving adaptability across different devices and screen configurations.

- **Bug Fixes**
  - Improved handling of touch controls, hit targets, DPI scaling, portrait and landscape orientations, and small-screen layouts.

- **Tests**
  - Added comprehensive automated coverage for layout scaling and screen configuration edge cases.

### CODE REVIEW

File: NEWS.txt [hunk 1]
```diff
@@ -146,6 +146,7 @@ Version 7.46 - not yet released
   - describe airspace fill as a transparent colour, without hatch
     patterns or the Airspace transparency setting
 * development
+  - tests: add TAP TestLayout for Screen/Layout DPI and scale math #3083
```
[the remark this answers is not in the document]
Comment (lordfolken) [reply NEWS.txt]: Removed. TAP coverage is not user-visible, so it does not belong in NEWS.txt.
